java-frontend/src/main/java/org/sonar/java/se/xproc/BehaviorCache.java

```diff
@@ -38,12 +39,18 @@
 
   private final SymbolicExecutionVisitor sev;
   private final SquidClassLoader classLoader;
+  private final SemanticModel semanticModel;
```

Comment:
Given the structure of the constructor this field can be null. 
It is only used to be passed to a bytecodeEGWalker which expects it to be non null. (No check of nullity are made on EGWalker side). 
I tend to think that the overloading of constructor is not required here (or at the very least should be annotated with @VisibleForTesting but for sake of simplicity I would avoid the overloading and update tests accordingly.